pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle/ecdhes_aead_composite_decrypt.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package subtle
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/tink/go/subtle/hybrid"
+)
+
+// ECDHESAEADCompositeDecrypt is an instance of ECDH-ES decryption with Concat KDF
+// and AEAD content decryption
+type ECDHESAEADCompositeDecrypt struct {
+	privateKey  *hybrid.ECPrivateKey
+	pointFormat string
+	encHelper   EncrypterHelper
+}
+
+// NewECDHESAEADCompositeDecrypt returns ECDH-ES composite decryption construct with Concat KDF/ECDH-ES key unwrapping
+// and AEAD payload decryption.
+func NewECDHESAEADCompositeDecrypt(pvt *hybrid.ECPrivateKey, ptFormat string,
+	encHelper EncrypterHelper) (*ECDHESAEADCompositeDecrypt, error) {
+	return &ECDHESAEADCompositeDecrypt{
+		privateKey:  pvt,
+		pointFormat: ptFormat,
+		encHelper:   encHelper,
+	}, nil
+}
+
+// Decrypt using composite ECDH-ES with a Concat KDF key unwrap and AEAD content decryption
+func (d *ECDHESAEADCompositeDecrypt) Decrypt(ciphertext, aad []byte) ([]byte, error) {
+	if d.privateKey == nil {
+		return nil, fmt.Errorf("ECDHESAEADCompositeDecrypt: missing recipient private key for key unwrapping")
+	}
+
+	keySize := d.encHelper.GetSymmetricKeySize()
+
+	var cek []byte
+
+	encData := new(EncryptedData)
+
+	err := json.Unmarshal(ciphertext, encData)
```

Comment:
so its a cipher text container?

Reply:
yes.. it's just a struct that contains crypto data to build a structured (JWE) message